core/listener_test.go

```diff
@@ -169,8 +169,13 @@ func createListener(
 ) *Listener {
 	p2pSub, err := p2p.NewSubscriber[*header.ExtendedHeader](ps, header.MsgID, p2p.WithSubscriberNetworkID(networkID))
 	require.NoError(t, err)
+
 	err = p2pSub.Start(ctx)
 	require.NoError(t, err)
+	err = p2pSub.SetVerifier(func(ctx context.Context, msg *header.ExtendedHeader) error {
+		return nil
+	})
+	require.NoError(t, err)
```

Comment:
Actually, looking at the fix made me realize that this is an actual API problem on Subscriber. Setting Verifier is optional, while Unregister on Stop happens all the time, causing this error. I think we should either document that setting verifier is a must or ignore the error Unregister as it was before so that the call is idempotent

Reply:
I completely agree.